### ISSUE TITLE
Suggestion: Based on Sender's Domain - add option to filter on TLD and ALL Sub-Domains instead

### ISSUE THREAD

**Issue by WoofGrrrr** [RealRaven2000/quickFilters#264]
When creating a new filter with "Based on Sender's Domain", add an option or some other way to filter on Top-Level-Domain
and all Sub-Domains.

It would create a new filter with "**Match any of the following**" selected.

For sender email address "abc@xxx.com" it would have the following two rules:
***
- From ends with "@xxx.com"
- From ends with ".xxx.com"
***

For sender email address "abc@def.xxx.com" it would have the SAME following two rules:
***
- From ends with "@xxx.com"
- From ends with ".xxx.com"
***

Companies in particular change the sub-domains they use to send emails constantly.  I use the above pattern all the time so that
I don't have to keep editing my filter rules to keep up with it.  It is a real pain!

**Comment by RealRaven2000** [comment]
You can already add such a thing. Make a new custom template:

on quickFilters settings advanced - enabled Custom Templates and click Create New:

![image](https://github.com/user-attachments/assets/2d4a2657-5d2a-43ed-a5c8-2a4d44f3e6cc)

Select the custom variable domain of email 2x, then add @ and . in front:

![image](https://github.com/user-attachments/assets/b1eb8091-fece-40a3-b8f4-f754feb36743)

Save the template - you can now create filters based on this.

**Comment by WoofGrrrr** [comment]
> You can already add such a thing. Make a new custom template:
> 
> on quickFilters settings advanced - enabled Custom Templates and click Create New:

I guess you have to be a paid user to create templates.

**Comment by RealRaven2000** [comment]
> > You can already add such a thing. Make a new custom template:
> > on quickFilters settings advanced - enabled Custom Templates and click Create New:
> 
> I guess you have to be a paid user to create templates.

That's right. I may add some rudimentary templates for "free as in beer" users in the future, but I do not want to complicate the UI too much for standard users. full disclosure: I live of Add-on development, it's a full time job. So I am not sponsored by Mozilla or donations, it just wouldn't scale enough to keep my Add-ons and my family alive at the same time.

**Comment by WoofGrrrr** [comment]
Completley understand

**Comment by WoofGrrrr** [comment]
I bought a license and created my first template.  Thanks.

I gather that "quickFilterCustomTemplate:" must stay in the template name???

I'm unclear as to what the "Actions" Check Boxes on the "Create Filters" Window do.

I thought that, when you click the "Create Filter..." button, the selected actions would be automatically added to the actions list on the "Filter Rules" Window.  For example, I checked "Tags", expecting to see "Tag Message" in the list, and then I would need to select which Tag to add.  But that didn't happen.

**Comment by RealRaven2000** [comment]
> I gather that "quickFilterCustomTemplate:" must stay in the template name???

Yes! because I do not store (or create) the template completely independently of Thunderbird, it "pretends" to be a real filter. And there is no unique id for filters, so I had to make it recognizable using the name. Don;'t remove the prefix or quickFilters will not know that it is a template.

> 
> I'm unclear as to what the "Actions" Check Boxes on the "Create Filters" Window do.

Do you mean these?

![image](https://github.com/user-attachments/assets/ac5d6b9d-e883-4963-8782-9b7e523d38e1)

most of this should be documented here:

https://quickfilters.quickfolders.org/index.html#options

If you have specific questions, please follow up.

> I thought that, when you click the "Create Filter..." button, the selected actions would be automatically added to the actions list on the "Filter Rules" Window. For example, I checked "Tags", expecting to see "Tag Message" in the list, and then I would need to select which Tag to add. But that didn't happen.

no, the actions are added automatically at the moment and it's a global settings in the assistant. I cannot add specifics like that in actions to the custom template dialog because it needs to  go through the standard THunderbird filter editor, and there is only so much I can "fake" in this editor. 

In order to add even more funcitonality we would need some custom attributes in the core architecture of filters, but I am interested what exactly you are looking for.

the main idea with custom filter templates is to conveniently prefill the search terms with data from the selected email when you run the wizard (see custom variables on the top right)

**Comment by WoofGrrrr** [comment]
Sorry for the ambiguity.   I meant the "Actions" Panel on the "Create Filters" Window:
 
![QF_Create-Filters](https://github.com/user-attachments/assets/91bcc08c-e927-4ac2-ba18-a3f6aef3b07a)

Here, I thought that if I checked "Star", for example, that an "Add Star" Action would be added to the Filter's actions list.  But this doesn't happen.  So I don't know what these checkboxes do.

**Comment by RealRaven2000** [comment]
> Here, I thought that if I checked "Star", for example, that an "Add Star" Action would be added to the Filter's actions list. But this doesn't happen. So I don't know what these checkboxes do.

I think the star will only be added if you add a star (or have a star) in the originating email,. At least that's what how the other ones workm it's based on the original mail; but I would need to double check that. It's the same with the tags / prority - they would/should be based on the mail that you use when you create the filter with the assistant.

**Comment by WoofGrrrr** [comment]
Ah!  OK.  Thanks.  I can just do it  with templates then.

**Comment by RealRaven2000** [comment]
> Ah! OK. Thanks. I can just do it with templates then.

You have to think about the way the assistant works:

for instance, you star a message or you add a specific tag (or multiple ones) - then you engage the assistant, right-click the message and right-click create filter from message:

![image](https://github.com/user-attachments/assets/22af2540-9b64-4316-a57d-3206827a52ef)

this will extract the data from the email and base the new filter actions (star? which tags?) on that... (including the target folder)

the assistant can currently be auto-triggered by moving / copying a mail and by tagging.

**Comment by WoofGrrrr** [comment]
Thanks for the clarification.
